crates/matrix-sdk/src/sliding_sync/room.rs

```diff
@@ -77,16 +78,18 @@ impl SlidingSyncRoom {
 
     fn timeline_builder(&self) -> Option<TimelineBuilder> {
         if let Some(room) = self.client.get_room(&self.room_id) {
-            let timeline_queue = self.timeline_queue.lock_ref().to_vec();
-            let prev_batch = self.prev_batch.lock_ref().clone();
-            Some(Timeline::builder(&room).events(prev_batch, timeline_queue))
+            Some(
+                Timeline::builder(&room)
+                    .events(self.prev_batch.get_cloned(), self.timeline_queue.lock_ref().to_vec()),
```

Comment:
`get_cloned` also locks, internally. But doesn't hurt 🤷🏼

Reply:
Ah OK, but yeah, it doesn't hurt.